### ISSUE TITLE
Fleet page no available upgrade 8.11.0

### ISSUE THREAD

**Issue by Aqualie** [elastic/elastic-agent#3728]
- Version: Elastic stack 8.11.0
- Operating System: K8S
- Discuss Forum URL:
- Steps to Reproduce:
- 1. Deploy Kibana and fleet server 8.11.0
- 2. Deploy elastic-agent with 8.10.4 and enroll into fleet
- 3. Kibana Fleet page does not show any upgradeable version to 8.11.0

**Comment by pierrehilbert** [comment]
@kilfoyle would you have to have a look here please?

**Comment by kilfoyle** [comment]
@pierrehilbert I'm happy to help but if I'm not mistaken isn't this a problem with the Fleet UI not showing the latest available upgrade, rather than a docs problem?

**Comment by pierrehilbert** [comment]
@kilfoyle My bad I read too quickly and thought it was something related to documentation.
@kpollich Should be able to have more insights then :-)

**Comment by kpollich** [comment]
Yeah I am seeing the same behavior, my 8.10.4 does not show as upgradeable

![image](https://github.com/elastic/elastic-agent/assets/6766512/1f13472e-3fd5-4e6d-95b8-5c71e8a2946f)

Diagnostics bundle for my agent here: 
[elastic-agent-diagnostics-2023-11-08T15-28-09Z-00.zip](https://github.com/elastic/elastic-agent/files/13298495/elastic-agent-diagnostics-2023-11-08T15-28-09Z-00.zip)

Contents of `.fleet-agents` document for this agent:

```json
{
        "_index": ".fleet-agents-7",
        "_id": "47e4bcbe-bbdb-46d9-8dc7-f345db325777",
        "_score": 1,
        "_source": {
          "access_api_key_id": "0wWFr4sBM7qpLiAtbYWc",
          "action_seq_no": [
            2
          ],
          "active": true,
          "agent": {
            "id": "47e4bcbe-bbdb-46d9-8dc7-f345db325777",
            "version": "8.10.4"
          },
          "enrolled_at": "2023-11-08T15:20:39Z",
          "local_metadata": {
            "elastic": {
              "agent": {
                "build.original": "8.10.4 (build: a92ca32b922dc62e45b532600a8562ee1c82205f at 2023-10-12 09:09:47 +0000 UTC)",
                "complete": false,
                "id": "47e4bcbe-bbdb-46d9-8dc7-f345db325777",
                "log_level": "info",
                "snapshot": false,
                "upgradeable": true,
                "version": "8.10.4"
              }
            },
            "host": {
              "architecture": "aarch64",
              "hostname": "agent1",
              "id": "2394f3b7e2994a918022af6fa0d735f9",
              "ip": [
                "127.0.0.1/8",
                "::1/128",
                "192.168.66.3/24",
                "fddf:ddd3:92f6:795d:5054:ff:fee6:f1c9/64",
                "fe80::5054:ff:fee6:f1c9/64"
              ],
              "mac": [
                "52:54:00:e6:f1:c9"
              ],
              "name": "agent1"
            },
            "os": {
              "family": "debian",
              "full": "Ubuntu jammy(22.04.3 LTS (Jammy Jellyfish))",
              "kernel": "5.15.0-87-generic",
              "name": "Ubuntu",
              "platform": "ubuntu",
              "version": "22.04.3 LTS (Jammy Jellyfish)"
            }
          },
          "policy_id": "07546000-4e7e-11ee-8460-1b348f164b1b",
          "type": "PERMANENT",
          "outputs": {
            "default": {
              "api_key": "1QWFr4sBM7qpLiAtfIX6:Z1ZTP0CyQ5i2MvVMc_v5Ww",
              "permissions_hash": "83820149229d9c3e2ca7acca19857102698746239bdf8275cacfb350fa61700f",
              "type": "elasticsearch",
              "api_key_id": "1QWFr4sBM7qpLiAtfIX6"
            }
          },
          "policy_revision_idx": 2,
          "policy_coordinator_idx": 1,
          "updated_at": "2023-11-08T15:29:45Z",
          "components": [
            {
              "id": "log-default",
              "units": [
                {
                  "id": "log-default-logfile-system-b8dc6063-2e9e-4111-ad00-9d2b4257075e",
                  "type": "input",
                  "message": "Healthy",
                  "status": "HEALTHY"
                },
                {
                  "id": "log-default-logfile-nginx-8572ee5d-e6ba-4d94-aa40-a835eac3c020",
                  "type": "input",
                  "message": "Healthy",
                  "status": "HEALTHY"
                },
                {
                  "id": "log-default",
                  "type": "output",
                  "message": "Healthy",
                  "status": "HEALTHY"
                }
              ],
              "type": "log",
              "message": "Healthy: communicating with pid '1581'",
              "status": "HEALTHY"
            },
            {
              "id": "system/metrics-default",
              "units": [
                {
                  "id": "system/metrics-default-system/metrics-system-b8dc6063-2e9e-4111-ad00-9d2b4257075e",
                  "type": "input",
                  "message": "Healthy",
                  "status": "HEALTHY"
                },
                {
                  "id": "system/metrics-default",
                  "type": "output",
                  "message": "Healthy",
                  "status": "HEALTHY"
                }
              ],
              "type": "system/metrics",
              "message": "Healthy: communicating with pid '1587'",
              "status": "HEALTHY"
            },
            {
              "id": "nginx/metrics-default",
              "units": [
                {
                  "id": "nginx/metrics-default-nginx/metrics-nginx-8572ee5d-e6ba-4d94-aa40-a835eac3c020",
                  "type": "input",
                  "message": "Healthy",
                  "status": "HEALTHY"
                },
                {
                  "id": "nginx/metrics-default",
                  "type": "output",
                  "message": "Healthy",
                  "status": "HEALTHY"
                }
              ],
              "type": "nginx/metrics",
              "message": "Healthy: communicating with pid '1592'",
              "status": "HEALTHY"
            },
            {
              "id": "filestream-monitoring",
              "units": [
                {
                  "id": "filestream-monitoring-filestream-monitoring-agent",
                  "type": "input",
                  "message": "Healthy",
                  "status": "HEALTHY"
                },
                {
                  "id": "filestream-monitoring",
                  "type": "output",
                  "message": "Healthy",
                  "status": "HEALTHY"
                }
              ],
              "type": "filestream",
              "message": "Healthy: communicating with pid '1600'",
              "status": "HEALTHY"
            },
            {
              "id": "beat/metrics-monitoring",
              "units": [
                {
                  "id": "beat/metrics-monitoring-metrics-monitoring-beats",
                  "type": "input",
                  "message": "Healthy",
                  "status": "HEALTHY"
                },
                {
                  "id": "beat/metrics-monitoring",
                  "type": "output",
                  "message": "Healthy",
                  "status": "HEALTHY"
                }
              ],
              "type": "beat/metrics",
              "message": "Healthy: communicating with pid '1608'",
              "status": "HEALTHY"
            },
            {
              "id": "http/metrics-monitoring",
              "units": [
                {
                  "id": "http/metrics-monitoring-metrics-monitoring-agent",
                  "type": "input",
                  "message": "Healthy",
                  "status": "HEALTHY"
                },
                {
                  "id": "http/metrics-monitoring",
                  "type": "output",
                  "message": "Healthy",
                  "status": "HEALTHY"
                }
              ],
              "type": "http/metrics",
              "message": "Healthy: communicating with pid '1615'",
              "status": "HEALTHY"
            }
          ],
          "last_checkin_message": "Running",
          "last_checkin_status": "online",
          "last_checkin": "2023-11-08T15:29:41Z"
        }
      }
```

Seems like `upgradeable` is true in the agent doc. Will continue investigating.

**Comment by WiegerElastic** [comment]
8.11.0 is offered through the UI, but VERY briefly. Manual upgrading of endpoints through Devtools seems to work. More info here: https://github.com/elastic/infosec/issues/15358

**Comment by juliaElastic** [comment]
This seems the same issue as https://github.com/elastic/kibana/issues/169825#issuecomment-1803491580

**Comment by kpollich** [comment]
Closing in favor of https://github.com/elastic/kibana/issues/169825 - please see responses and track progress on that issue. Targeting 8.11.1 for a fix here.

**Comment by kpollich** [comment]
Hi folks, I'm following up on all the issues/threads where this was reported and tracked. 

Due to 8.11.1 being pulled forward for an emergency release, this bugfix has missed the 8.11.1 build candidate window. The root cause fix in https://github.com/elastic/kibana/pull/170974 will be available in the 8.11.2 release scheduled for public availability on December 4th, 2023. 

In the meantime, there is a documented workaround available in our support knowledge base here: https://support.elastic.co/knowledge/86bef2c1 (requires an Elastic Cloud login).

**Comment by WiegerElastic** [comment]
> Hi folks, I'm following up on all the issues/threads where this was reported and tracked.
> 
> Due to 8.11.1 being pulled forward for an emergency release, this bugfix has missed the 8.11.1 build candidate window. The root cause fix in [elastic/kibana#170974](https://github.com/elastic/kibana/pull/170974) will be available in the 8.11.2 release scheduled for public availability on December 4th, 2023.
> 
> In the meantime, there is a documented workaround available in our support knowledge base here: [support.elastic.co/knowledge/86bef2c1](https://support.elastic.co/knowledge/86bef2c1) (requires an Elastic Cloud login).

I find it a bit concerning that we are letting customers sit with this bug for 4+ weeks, even with a workaround present. It's really not a good look.

**Comment by kpollich** [comment]
> I find it a bit concerning that we are letting customers sit with this bug for 4+ weeks, even with a workaround present. It's really not a good look.

I agree and raised the same in the emergency release thread and was rebuffed, unfortunately. The build candidate was built in a hurry over the weekend and the release team was adamant on getting it out ASAP today.
